Sources/SymbolKit/UnifiedSymbolGraph/UnifiedSymbolGraph.swift

```diff
@@ -160,6 +160,190 @@ extension UnifiedSymbolGraph {
+                """
+                UnifiedSymbolGraph.combineOverloadGroups found a candidate overload group that didn't
+                match any existing overload group from its constituent symbol graphs.
+
+                The unified graph found that \(firstOverload.uniqueIdentifier) should be the default
+                overload, but the constituent symbol graphs contained the following overload groups:
+                \(siblingOverloadGroups)
+                """
+                )
+                continue
+            }
+
+            for overloadIndex in sortedOverloads.indices {
+                let overloadedSymbol = sortedOverloads[overloadIndex]
+
+                // 3. Ensure that all the overloaded symbols have `overloadOf` relationships pointing
+                // to the computed overload group.
+                for selector in overloadedSymbol.mainGraphSelectors {
+                    let newRelationship = SymbolGraph.Relationship(
+                        source: overloadedSymbol.uniqueIdentifier,
+                        target: computedOverloadGroup,
+                        kind: .overloadOf,
+                        targetFallback: nil)
+                    if relationshipsByLanguage[selector]?.contains(where: { otherRelationship in
+                        otherRelationship.source == newRelationship.source &&
+                        otherRelationship.target == newRelationship.target &&
+                        otherRelationship.kind == newRelationship.kind
+                    }) != true {
+                        relationshipsByLanguage[selector, default: []].append(newRelationship)
+                    }
+                }
+
+                // 4. Create a new OverloadData instance and save it to the overloaded symbol.
+                let overloadData = SymbolGraph.Symbol.OverloadData(
+                    overloadGroupIdentifier: computedOverloadGroup,
+                    overloadGroupIndex: overloadIndex)
+                overloadedSymbol.unifiedMixins[SymbolGraph.Symbol.OverloadData.mixinKey] = overloadData
+            }
+
+            // 5. If there were any other overload groups that pointed to the same symbols, remove
+            // them from the unified graph, along with any relationships that included them.
+            for siblingOverloadGroup in siblingOverloadGroups {
+                // Remove all the sibling groups from the dictionary we're iterating over so we don't
+                // recalculate all this all over again.
+                overloadGroups.removeValue(forKey: siblingOverloadGroup)
+
+                if siblingOverloadGroup != computedOverloadGroup {
+                    symbols.removeValue(forKey: siblingOverloadGroup)
+                }
+            }
+            siblingOverloadGroups.remove(computedOverloadGroup)
+            for selector in relationshipsByLanguage.keys {
+                relationshipsByLanguage[selector]?.removeAll(where: { relationship in
```

Comment:
Same here - if we have 10,000s of symbols will this be very slow?

Reply:
We need to clean up the relationships listings somehow, and like i said, the type signature here is public API. There can probably be improvements made to iterate this collection fewer times (performing the "make sure the computed overload group references all the symbols" check at the same time as the "remove the relationships for the extra overload groups" pass, combining the "extra overload groups" passes together, etc) but iterating the collection is unavoidable.